### ISSUE TITLE
show screenshots of hibernated pages

### ISSUE THREAD

**Issue by 343max** [343max/TabHibernation#8]

**Comment by HorayNarea** [comment]
i tried to implement this via chrome.tabs.captureVisibleTab but that only takes a screenshot of (surpise surprise!) the visible(==focused==active) tab
one could make a screenshot-loop that sets every tab as active, takes a screenshot and sets the tab as inactive again but this means a lot of tab switching and "flickering" and probably will slow down massively with many tabs

i also looked into (weird) alternative methods like chrome.tabCapture but it either didn't work at all or was unreliable and slow as hell

**in short: at the current state of Chrome-APIs i think this feature is impossible (or at least very hard) while still having a nice user-experience**

**Comment by HorayNarea** [comment]
untested idea: we might inject JS into every page that makes a screenshot via captureVisibleTab() when the tab gets focus and use it when hibernating this tab
(but this could make the extension a bit memory-heavy which would make it kind of pointless)

**Comment by 343max** [comment]
I tried it a few times but it seems to be impossible for security reasons.

**Comment by elgalu** [comment]
I found 2 extensions that are able to do this, perhaps you can try to investigate how:
- [Tab Suspender](https://chrome.google.com/webstore/detail/tab-suspender/fiabciakcmgepblmdkmemdbbkilneeeh?hl=en) was quite slow for my taste.
- [Tab Hibernate](https://chrome.google.com/webstore/detail/tab-hibernate/ammlihljcndoijbkoobiobhjgoopiidn?hl=en) seems to be working faster, note screen shots are optional via checkbox.

Even though there seems to be not much saving in memory what is also interesting is to avoid having all those background JS scripts running on the tabs which peaks the CPU ;)

**Comment by HorayNarea** [comment]
Thanks for the input, I will look into those extensions!
And I agree, a minimum CPU usage (optimally zero) should be a priority
